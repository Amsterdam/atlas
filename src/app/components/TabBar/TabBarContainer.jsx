import { connect } from 'react-redux';
import TabBar from './TabBar';
import { getSearchQuery } from '../../../shared/ducks/data-search/selectors';
import { toDatasets } from '../../../store/redux-first-router';

const mapStateToProps = (state) => ({
  searchQuery: getSearchQuery(state)
});

const mapDispatchToProps = (dispatch) => ({
  goToDatasets: () => dispatch(toDatasets())
});

export default connect(mapStateToProps, mapDispatchToProps)(TabBar);
