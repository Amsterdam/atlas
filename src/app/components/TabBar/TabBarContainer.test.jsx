import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import TabBarContainer from './TabBarContainer';

const initialState = {
  dataSearch: {
    // geoSearch: {}
    query: 'dam'
  }
};

describe('TabBarContainer', () => {
  it('should render the container with the basic properties', () => {
    const store = configureMockStore()({ ...initialState });
    const props = {
      // searchQuery: 'dam'
    };
    const component = shallow(<TabBarContainer {...props} />, { context: { store } });
    expect(component).toMatchSnapshot();
  });
});
