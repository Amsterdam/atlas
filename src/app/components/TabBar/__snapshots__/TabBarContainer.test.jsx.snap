// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarContainer should render the container with the basic properties 1`] = `
<TabBar
  goToDatasets={[Function]}
  searchQuery="dam"
  showDatasetsButton={false}
  totalNumberOfResults={1}
/>
`;
